src/shared/sam/debugger/csharpSamDebug.ts

```diff
@@ -100,40 +95,69 @@ async function ensureDebuggerPathExists(debuggerPath: string): Promise<void> {
-): Promise<void> {
+async function _installDebugger({ debuggerPath, channelLogger }: InstallDebuggerArgs): Promise<void> {
     await ensureDebuggerPathExists(debuggerPath)
 
     try {
-        const samCliContext = getSamCliContext()
-        const samCliVersionValidatorResult = await samCliContext.validator.getVersionValidatorResult()
-        const samCliVersion = samCliVersionValidatorResult.version
-
-        const imageStr = getSamCliDockerImageName(samCliVersion, lambdaRuntime)
-
         channelLogger.info(
             'AWS.samcli.local.invoke.debugger.install',
-            'Installing .NET Core Debugger to {0} using Docker image {1}...',
-            debuggerPath,
-            imageStr
+            'Installing .NET Core Debugger to {0}...',
+            debuggerPath
         )
 
-        await dockerClient.invoke({
-            command: 'run',
-            image: imageStr,
-            removeOnExit: true,
-            mount: {
-                type: 'bind',
-                source: debuggerPath,
-                destination: '/vsdbg',
-            },
-            entryPoint: {
-                command: 'bash',
-                args: ['-c', 'curl -sSL https://aka.ms/getvsdbgsh | bash /dev/stdin -v latest -l /vsdbg'],
-            },
+        const vsDbgVersion = 'latest'
+        const vsDbgRuntime = 'linux-x64'
+
+        const installScriptPath = await downloadInstallScript(debuggerPath)
+
+        let installCommand: string
+        let installArgs: string[]
+        if (os.platform() == 'win32') {
+            installCommand = `${process.env['WINDIR']}\\System32\\WindowsPowerShell\\v1.0\\powershell.exe`
+            installArgs = [
+                '-NonInteractive',
+                '-NoProfile',
+                '-WindowStyle',
+                'Hidden',
+                '-ExecutionPolicy',
+                'RemoteSigned',
+                '-File',
+                installScriptPath,
+                '-Version',
+                vsDbgVersion,
+                '-RuntimeID',
+                vsDbgRuntime,
+                '-InstallPath',
+                debuggerPath,
+            ]
+        } else {
+            await new ChildProcess('chmod', {}, 'u+x', installScriptPath).run()
+
+            installCommand = installScriptPath
+            installArgs = ['-v', vsDbgVersion, '-r', vsDbgRuntime, '-l', debuggerPath]
+        }
+
+        const childProcess = new ChildProcess(installCommand, {}, ...installArgs)
+
+        const installPromise = new Promise<void>(async (resolve, reject) => {
+            await childProcess.start({
+                onStdout: (text: string) => {
+                    channelLogger.channel.append(text)
+                },
+                onStderr: (text: string) => {
+                    channelLogger.channel.append(text)
+                },
+                onClose(code: number) {
+                    if (code) {
+                        reject(`Exited with code ${code}`)
```

Comment:
10x more useful error by including the command in the error message, something like:

```suggestion
                        reject(`command failed (exit code: ${code}): ${installCommand}`)
```

Reply:
Thanks, reads better